src/UBAFeeCalculator/UBAFeeSpokeCalculatorAnalog.ts

```diff
@@ -138,12 +138,17 @@ export function getEventFee(
 
   // Next, we'll need to compute the first balancing fee from the running balance of the spoke
   // to the running balance of the spoke + the amount
-  const balancingFee = computePiecewiseLinearFunction(
+  let balancingFee = computePiecewiseLinearFunction(
     flowCurve,
     runningBalance,
     amount.mul(flowType === "inflow" ? 1 : -1)
   );
 
+  // We need to account for the balancing fee being greater than the incentive balance
+  if (balancingFee.gt(lastValidatedIncentiveRunningBalance)) {
+    balancingFee = balancingFee.mul(config.getUbaRewardMultiplier()).div(fixedPointAdjustment);
```

Comment:
This is not correct. This should implement:

```
   discountFactor = min(1, (balancingFee - incentiveBalance) / balancingFee)
   balancingFee *= 1 - discountFactor
```

Reply:
```ts
// This should never error. `getUbaRewardMultiplier` should default to 1
balancingFee = balancingFee.mul(config.getUbaRewardMultiplier())

// We need to account for the balancing fee being greater than the incentive balance
  if (balancingFee.gt(lastValidatedIncentiveRunningBalance)) {
      const discountFactor = min(BigNumber.from(1), (balancingFee.sub(incentiveBalance).div(balancingFee)))
      balancingFee = balancingFee.mul(BigNumber.from(1).sub(discountFactor))
  }
```

Reply:
Resolved here: https://github.com/across-protocol/sdk-v2/pull/290/commits/53f751c78854fb21fd7e386d7a6e9a656034e262